src/FlatSharp/TypeModel/Vectors/UnityNativeArrayVectorTypeModel.cs

```diff
@@ -0,0 +1,97 @@
+/// </summary>
+public class UnityNativeArrayVectorTypeModel : BaseVectorTypeModel
+{
+    internal UnityNativeArrayVectorTypeModel(Type vectorType, TypeModelContainer provider) : base(vectorType, provider)
+    {
+    }
+
+    public override string LengthPropertyName => nameof(Memory<byte>.Length);
+
+    public override Type OnInitialize()
+    {
+        return this.ClrType.GetGenericArguments()[0];
+    }
+    
+    public override void Validate()
+    {
+        if (!this.ClrType.IsGenericType || this.ClrType.GetGenericTypeDefinition().FullName != "Unity.Collections.NativeArray`1")
+            throw new InvalidFlatBufferDefinitionException(
+                $"UnityNativeArray vectors must be of type Unity.Collections.NativeArray. Type = {this.GetCompilableTypeName()}.");
+
+        var genericArgumentType = this.ClrType.GetGenericArguments()[0];
+        if (ItemTypeModel.SchemaType != FlatBufferSchemaType.Scalar && ItemTypeModel.SchemaType != FlatBufferSchemaType.Struct)
+            throw new InvalidFlatBufferDefinitionException(
+                $"UnityNativeArray vectors only support scalar or struct generic arguments. Type = {this.GetCompilableTypeName()}.");
+        
+        if (!genericArgumentType.IsValueType)
+            throw new InvalidFlatBufferDefinitionException(
+                $"UnityNativeArray vectors only support value type generic arguments. Type = {this.GetCompilableTypeName()}.");
+        
+        if (ItemTypeModel.PhysicalLayout.Length != 1)
+            throw new InvalidFlatBufferDefinitionException(
+                $"UnityNativeArray vectors do not support the specified generic argument type. Type = {this.GetCompilableTypeName()}.");
+
+
+        base.Validate();
+    }
+
+    protected override string CreateLoop(FlatBufferSerializerOptions options, string vectorVariableName, string numberOfItemsVariableName, string expectedVariableName, string body)
+    {
+        FlatSharpInternal.Assert(false, "Not expecting to do loop get max size for memory vector");
+        throw new Exception();
+    }
+
+    public override CodeGeneratedMethod CreateParseMethodBody(ParserCodeGenContext context)
+    {
+        ValidateWriteThrough(
+            writeThroughSupported: false,
+            this,
+            context.AllFieldContexts,
+            context.Options);
+
+        if (context.Options.Lazy)
+        {
+            string body = $@"
+                if (!buffer.IsPinned)
+                    throw new NotSupportedException(""Lazy parsing of a NativeArray requires a pinned buffer."");
+                var bufferSpan = {context.InputBufferVariableName}.UnsafeReadSpan<{context.InputBufferTypeName}, {this.ItemTypeModel.GetGlobalCompilableTypeName()}>({context.OffsetVariableName});
+                return Unity.Collections.LowLevel.Unsafe.NativeArrayUnsafeUtility.ConvertExistingDataToNativeArray<{this.ItemTypeModel.GetGlobalCompilableTypeName()}>(bufferSpan, Allocator.None);
```

Comment:
Sadly, you'll have to also emit the ENABLE_COLLECTIONS_CHECKS #ifdef and the manual safety handle setting, despite Alloactor.None.  This is a super annoying wart with this API.   (This should crash if you try to read from one of these returned arrays in the editor or in development player builds currently.)

e.g.:

```
                   var foo = NativeArrayUnsafeUtility.ConvertExistingDataToNativeArray<>(...);

  #if ENABLE_UNITY_COLLECTIONS_CHECKS
                      NativeArrayUnsafeUtility.SetAtomicSafetyHandle(ref foo, AtomicSafetyHandle.GetTempUnsafePtrSliceHandle());
  #endif
                       return foo;
```

But I don't think TempUnsafePtrSliceHandle is correct. (It'll work fine for now.)  If we want to be _totally_ correct, we can allocate a safety handle that corresponds to the original buffer we're creating the array from, and use that.  That way if the input buffer goes away for some reason and someone kept a NativeArray around, they'll get a proper error.